e2e_tests/tests/experiment/test_noop.py

```diff
@@ -243,3 +243,139 @@ def test_startup_hook() -> None:
     exp.run_basic_test(
         conf.fixtures_path("no_op/startup-hook.yaml"), conf.fixtures_path("no_op"), 1,
     )
+
+
+@pytest.mark.e2e_cpu  # type: ignore
+def test_rng_restore() -> None:
+    experiment = exp.run_basic_test(
+        conf.fixtures_path("pytorch-rng-saver/const.yaml"),
+        conf.fixtures_path("pytorch-rng-saver"),
+        1,
+    )
+
+    trials = exp.experiment_trials(experiment)
+    first_trial = trials[0]
+
+    assert len(trials) == 1
+    assert len(first_trial["steps"]) == 3
+
+    first_step = first_trial["steps"][0]
+    second_checkpoint_id = first_step["checkpoint"]["id"]
+
+    config_base = conf.load_config(conf.fixtures_path("pytorch-rng-saver/const.yaml"))
+    config_obj = copy.deepcopy(config_base)
+    config_obj["searcher"]["source_checkpoint_uuid"] = first_step["checkpoint"]["uuid"]
+
+    experiment2 = exp.run_basic_test_with_temp_config(
+        config_obj, conf.fixtures_path("pytorch-rng-saver"), 1
+    )
+
+    trials = exp.experiment_trials(experiment2)
+    second_trial = trials[0]
```

Comment:
probably just `trial_2 = exp.experiment_trials(experiment2)[0]`

Reply:
I need `trials` for one of the assertion statements: `assert len(trials) == 1`. Is this unnecessary?

Reply:
Yeah I don't necessary, honestly.  We have other tests with those sorts of checks.